environments/fetcher/fetcher.go

```diff
@@ -254,6 +254,15 @@ func (fetcher *Fetcher) Fetch(req FetchRequest) (int, error) {
 		if req.FetchType == FETCH_SOURCE {
 			archive = &pkg.Spec.Source
 		} else if req.FetchType == FETCH_DEPLOYMENT {
+			// sometimes, the user may issue curl request to the function even before the source code is built into a deploy pkg.
```

Comment:
small nit: "may issue curl request" -> "may invoke the function" (since there are other ways to invoke functions besides http)